src/contract.py

```diff
@@ -208,30 +211,21 @@ def __init__(self, barrier_level: float, up_down: UpDown, in_out: InOut) -> None
             self.raise_incorrect_in_out_type()
         self.in_out: InOut = in_out
 
-    def get_barrier_level(self) -> float:
-        return self._barrier_level
-
-    def get_up_down(self) -> UpDown:
-        return self._up_down
-
-    def get_in_out(self) -> InOut:
-        return self._in_out
-
-    def is_breached(self, spot: dict[float, float], vol: float = -1) -> float:
+    def is_breached(self, spot: dict[float, float], vol: float) -> float:
         timeline = list(spot.keys())
         observations = [spot[t] for t in timeline]
-        if vol == -1:
-            if self._up_down == UpDown.UP:
-                return float(any([self._barrier_level <= price for price in observations]))
-            elif self._up_down == UpDown.DOWN:
-                return float(any([self._barrier_level >= price for price in observations]))
+        if np.isnan(vol):      # standard case without probabilities
+            if self.up_down == UpDown.UP:
+                return float(any([self.barrier_level <= price for price in observations]))
+            elif self.up_down == UpDown.DOWN:
+                return float(any([self.barrier_level >= price for price in observations]))
             else:
                 self._raise_incorrect_up_down_type()
```

Comment:
No need for underscore prefix.

Reply:
deleted, also from the raise_unsupported_greek_method_error